### PR TITLE
Complete overhaul of dashboard navigation

### PR DESCRIPTION
**Pull request by timcheadle**
To make the site easier to use, more flexible, and make better use of space, @dburka overhauled the whole navigation. I've updated all the specs to match.

### CODE REVIEW

**Comment by timcheadle** [comment]
@harimohanraj89 I've reviewed this and it looks solid. It's almost all @dburka's work, but since I created the PR I can't approve it.

**Comment by harimohanraj89** [comment]
@dburka Login panel is wider on fullscreen than before. Intentional?

File: app/assets/stylesheets/partials/_navigation.scss [hunk 1]
```diff
@@ -2,52 +2,250 @@
...
+    color: $blue-mid;
+
+    &:hover {
+      background: $blue-dark;
+    }
+  }
+
+  .nav-link.active {
+    i {
+      color: $white !important;
```
Comment (harimohanraj89) [inline app/assets/stylesheets/partials/_navigation.scss:92]: Can we take a stab at not needing so many `!important`s in this file?

File: app/views/layouts/my_facilities.html.erb [hunk 2]
```diff
@@ -1,30 +1,4 @@
...
-          <%= link_to(my_facilities_drug_stocks_path(preserve_query_params(request.query_parameters, ["facility_group", "zone", "size", "for_end_of_month"])), class: "sub-nav-link #{active_action?("drug_stocks")}") do %>
-            Drug stock
-          <% end %>
-          <%= link_to(my_facilities_drug_consumption_path(preserve_query_params(request.query_parameters, ["facility_group", "zone", "size", "for_end_of_month"])), class: "sub-nav-link #{active_action?("drug_consumption")}") do %>
-            Drug consumption
-          <% end %>
-        <% end %>
-      </div>
-    </div>
-  </div>
```
Comment (harimohanraj89) [inline app/views/layouts/my_facilities.html.erb:27]: Looks like we don't need this layout anymore. Mind if I rip it out?

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
My review below. Could you test thoroughly that the right nav elements show up for the right access levels of users? We already have the following test users set up for easy testing:

* admin@simple.org - can do everything
* cvho@simple.org - manages multiple districts
* district_official@simple.org - views a district
* medical_officer@simple.org - can manage a facility i think?

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -30,12 +30,12 @@
     <%= stylesheet_link_tag "application", media: "all" %>
     <%= stylesheet_link_tag "print", media: "print" %>
     <%= javascript_include_tag "application" %>
+    <script src="https://kit.fontawesome.com/72c701f5ae.js" crossorigin="anonymous"></script>
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/views/layouts/application.html.erb:33]: I removed the CSS version of the FontAwesome kit and use the now-preferred JS method of including it. This is how we reference all of the icons we're using in the Dashboard. Is there a better way to do this?

File: app/assets/stylesheets/partials/_navigation.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (dburka) [reply app/assets/stylesheets/partials/_navigation.scss:92]: Got rid of all but one `!important` Great nudge @harimohanraj89

File: app/assets/stylesheets/partials/_dashboard.scss [hunk 4]
```diff
@@ -535,4 +624,76 @@ em.bg-orange {
   border-color: $orange-light !important;
 }
 
-.tl-fixed { table-layout: fixed; }
+.tl-fixed {
+  table-layout: fixed;
+}
+
+.resource-link {
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/assets/stylesheets/partials/_dashboard.scss:631]: Done

File: app/views/admin/facilities/_form.html.erb [hunk 5]
```diff
@@ -50,9 +50,6 @@
               Medical officers will receive teleconsult requests via WhatsApp at the number listed in their profile.
               Edit the user’s profile to update the number.
             </div>
-            <div class="mo-search">
-              <%= render "shared/user_teleconsult_search", facility_group_id: facility_group.id %>
-            </div>
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/views/admin/facilities/_form.html.erb:55]: Yikes. This was a mistake. Great catch. Added back.

File: app/views/layouts/application.html.erb [hunk 6]
```diff
@@ -44,6 +44,19 @@
   <body>
     <% if email_authentication_signed_in? %>
       <%= render "shared/nav_bar" %>
+
+      <% if current_admin.accessible_facilities(:view_reports).any? %>
+        <% if active_controller_and_action?(["my_facilities", "reports/regions"], ["index", "show", "details", "cohort"]) %>
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/views/layouts/application.html.erb:49]: Yes, please clean up @harimohanraj89

File: app/views/layouts/application.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply app/views/layouts/application.html.erb:33]: Nope this sounds great. Was wondering if this was just the new recommended way, or whether there were other constraints that were forcing us to move to kits.

File: app/views/shared/_region_search_script.html.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    input = $(event.target).find(".typeahead-input");
+    search(input);
+    return false;
+  });
+
+  form.on("keyup", function(event) {
+    search(event.target);
+    return false;
+  });
+</script>
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/views/shared/_region_search_script.html.erb:15]: Can you please help so this gets done properly @harimohanraj89 ?

[the remark this answers is not in the document]
Comment (harimohanraj89) [reply app/views/shared/_region_search_script.html.erb:15]: Yup totally 👍

File: app/views/shared/my_facilities/_inactive_facilities.html.erb [hunk 8]
```diff
@@ -4,7 +4,7 @@
       Inactive facilities
     </h2>
     <p>
-      Facilities with &lt;10 patients with a BP taken in the last 7 days
+      Facilities with &lt;10 patients with a BP taken
```
[the remark this answers is not in the document]
Comment (dburka) [reply app/views/shared/my_facilities/_inactive_facilities.html.erb:7]: The description is inaccurate and has been inaccurate for a long time. We actually show # in 7 days in one column and # in 30 days in the next column. The headers for the table make this clear.

File: app/views/layouts/my_facilities.html.erb (hunk 2 above)
Comment (dburka) [reply app/views/layouts/my_facilities.html.erb:27]: Yes, please tear it out!

File: app/views/shared/my_facilities/_inactive_facilities.html.erb (hunk 8 above)
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply app/views/shared/my_facilities/_inactive_facilities.html.erb:7]: Ha amazing. 💯

**Review by harimohanraj89 — APPROVED** [review]

**Pull request merged**